### PR TITLE
Add ac transit api services

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request introduces new dependencies and extensive test/mocking infrastructure to support AC Transit and GTFS Realtime API integrations. It adds environment variables for configuration, mock data generators, and comprehensive unit tests for the AC Transit Realtime service. These changes improve test coverage, reliability, and maintainability for transit data features.

**Environment and Dependency Additions**

* Added new dependencies: `gtfs-realtime-bindings`, `node-fetch`, and `tiny-invariant` to support GTFS Realtime API parsing and robust data fetching.
* Updated environment schema and mock environment (`src/env.ts`, `src/mocks/env.ts`) to include AC Transit and GTFS Realtime API URLs, tokens, polling intervals, and cache TTLs for various transit data types. [[1]](diffhunk://#diff-00ee6a506727e42b3fb03361b8216c805ede24dc3985d237de852a73bda59f30R10-R38) [[2]](diffhunk://#diff-a9216e62503c468bf15df4e9a767754780b00615f01880cf0f71292439ea1192R10-R23)
* Updated workspace configuration to build `protobufjs` only when needed, optimizing dependency management for GTFS Realtime.

**Testing and Mocking Infrastructure**

* Added mock data generators for AC Transit Realtime API responses, including bus stop profiles, predictions, vehicle positions, and system time, to facilitate reliable and repeatable unit tests.
* Added mock data generators for GTFS Realtime feed messages, entities (vehicle, trip update, alert), and array buffer serialization, supporting robust GTFS Realtime feature testing.

**Unit Test Coverage**

* Added comprehensive unit tests for the AC Transit Realtime service, covering bus stop profiles, predictions, vehicle positions, and system time, including error handling and edge cases.

### CODE REVIEW

File: src/services/actRealtime.ts [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+            }
+        } catch (error) {
+            // Fallback to server time on error
+            timestampMs = now;
+            console.error(
+                `Error fetching AC Transit system time: ${error instanceof Error ? error.message /* v8 ignore next */ : error}`
+            );
+        }
+
+        return new Date(timestampMs);
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/services/actRealtime.ts:375]: this endpoint returns in unix milliseconds https://api.actransit.org/transit/Help/Api/GET-actrealtime-time_unixTime_callback
